src/lib.rs

```diff
@@ -266,18 +266,31 @@ pub mod util {
     }
 
     pub fn load_private_key(filename: &str) -> Result<rustls::PrivateKey> {
-        let keyfile = fs::File::open(filename).map_err(|e| Error::Io(e))?;
-        let mut reader = BufReader::new(keyfile);
-        let mut keys = pemfile::rsa_private_keys(&mut reader)
-            .map_err(|_| Error::BadKey)?;
-
-        // Ensure there's only one key in the file.
-        if keys.len() != 1 {
-            return Err(Error::BadKeyCount);
+        use std::io::Seek;
+        let start = io::SeekFrom::Start(0);
+
+        fn expect_one(mut keys: Vec<rustls::PrivateKey>) -> Result<rustls::PrivateKey> {
```

Comment:
`expect` usually implies that a `panic!` might take place. Let's rename this to `take_one` or something of that sort.